protocol/contracts/libraries/LibUserBalance.sol

```diff
@@ -0,0 +1,196 @@
+/*
+ SPDX-License-Identifier: MIT
+*/
+
+pragma solidity ^0.7.6;
+pragma experimental ABIEncoderV2;
+
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/SafeCast.sol";
+import "./LibAppStorage.sol";
+import "./LibUniswap.sol";
+import "../interfaces/IBean.sol";
+import "../interfaces/IWETH.sol";
+
+/**
+ * @author LeoFib
+ * @title UserBalance Library handles internal read/write functions for Internal User Balances.
+**/
+
+library LibUserBalance {
+    
+  using SafeMath for uint256;
+  using SafeCast for uint256;
+
+  /**
+  * @dev Emitted when a user's Internal Balance changes, through interacting using Internal Balance.
+  *
+  */
+  event InternalBalanceChanged(address indexed user, IERC20 indexed token, int256 delta);
+
+  /**
+  * @dev Emitted when a user's ERC20 allowance is used to transfer tokens to an external account.
+  */
+  event ExternalBalanceTransfer(IERC20 indexed token, address indexed sender, address recipient, uint256 amount);
+
+  event BeanAllocation(address indexed account, uint256 beans);
+
+  function getInternalBalances(address account, IERC20[] memory tokens)
+    internal
+    view
+    returns (uint256[] memory balances)
+  {
+    balances = new uint256[](tokens.length);
+    for (uint256 i = 0; i < tokens.length; i++) {
+      balances[i] = _getInternalBalance(account, tokens[i]);
+    }
+  }
+
+  function getCombinedInternalExternalBalance(address account, IERC20 token)
+    internal
+    view
+    returns (uint256 combined_balance)
+  {
+    combined_balance = token.balanceOf(account).add(_getInternalBalance(account, token));
+    return combined_balance;
+  }
+
+  function _transferToExternalBalance(
+    IERC20 token,
+    address sender,
+    address recipient,
+    uint256 amount, 
+    bool fromInternalBalance
+  ) internal {
+    if (amount > 0) {
+      if (fromInternalBalance) _decreaseInternalBalance(sender, token, amount, true);
```

Comment:
This was my bad @Beasley9  Can you fix this so that it is false not true for allow partial